src/App.jsx

```diff
@@ -0,0 +1,187 @@
+import { useState, useEffect } from "react";
+import useFetchWeather from "./hooks/useFetchWeather";
+import TodayWeather from "./components/TodayWeather/TodayWeather";
+import LocationSearch from "./components/LocationSearch/LocationSearch";
+import RoundButton from "./components/RoundButton/RoundButton";
+import WeatherCard from "./components/WeatherCard/WeatherCard";
+import HighlightCard from "./components/HighlightCard/HighlightCard";
+import Skeleton from "./components/Skeleton/Skeleton";
+import { WEATHER_TYPES, UNITS } from "./constants";
+import "./App.css";
+
+function App() {
+  const initialLocation = "Firenze";
+  const {
+    currentWeatherData,
+    forecastWeatherData,
+    error,
+    unit,
+    isLoading,
+    setUnit,
+    fetchWeather,
```

Comment:
Instead of passing a `fetchWeather` and then use 2 fetchs to different places, let's use the `fetchWeatherForLocation`, which will be a combined fetch for both current and forecast.

```suggestion
    fetchWeatherForLocation,
```